src/interaction/inspector.jl

```diff
@@ -841,6 +851,10 @@ function show_data(inspector::DataInspector, plot::BarPlot, idx)
         a.indicator_visible[] = true
     end
 
+    pos = plot[1][][idx]
+    if plot.direction[] === :x
+        pos = reverse(pos)
+    end
```

Comment:
I'm not sure if the data should be flipped for user defined functions when the direction is horizontal. I assume direction is expected to be a visual change only. In that case, maybe the code should be:
```
    pos = plot[1][][idx]
    if to_value(get(plot, :inspector_label, automatic)) == automatic
        if plot.direction[] === :x
            pos = reverse(pos)
        end
        tt.text[] = position2string(pos)
    else
        tt.text[] = plot[:inspector_label][](plot, idx, pos)
    end
```

One can also make a case for never reversing the data, even in the automatic case. Curious to hear what the devs think.

Reply:
I generally thought of those positions as positions in the actual Axis/world. So if the bars end up at `reverse(pos)` with `direction[] === :x` I'd swap it for both.

Reply:
Ok, the PR currently swaps it for both so I'll leave it as is.
```
    pos = plot[1][][idx]
    if plot.direction[] === :x
        pos = reverse(pos)
    end
    if to_value(get(plot, :inspector_label, automatic)) == automatic
        tt.text[] = position2string(pos)
    else
        tt.text[] = plot[:inspector_label][](plot, idx, pos)
    end
```

Reply:
> I generally thought of those positions as positions in the actual Axis/world.

Actually, does that mean that `pos` should be the tooltip/bar position instead of the data point? Right now `pos` is the defined as the data point. The data point is different from the bar position in the stacked/dodged case.

Reply:
I need to correct myself a bit. At some point we changed this to be in the plots input space rather than world space (after model, transform_func) because that seems to be more useful with things like log plots.
I'm not sure what exactly the position should be. For most plots it's the position the tooltip points at and that's the only thing that makes sense. Here it's less clear. I'd lean towards it being that but I think updating the default label to work with stacked barplots is more important here.

Reply:
I think I understand. To confirm, the code that fixes label **location** is actually the code above:
```
    bbox = plot.plots[1][1][][idx]
    tt_frac = if plot.direction[] === :y
        Point2f(0.5, 1.0)
    else
        Point2f(1.0, 0.5)
    end
    tt_pos = fractionpoint(bbox, tt_frac)
    proj_pos = shift_project(scene, apply_transform_and_model(plot, to_ndim(Point3f, tt_pos, 0)))
```

The code here just specifies the label **data**:
```
    pos = plot[1][][idx]
    if plot.direction[] === :x
        pos = reverse(pos)
    end
    if to_value(get(plot, :inspector_label, automatic)) == automatic
        tt.text[] = position2string(pos)
    else
        tt.text[] = plot[:inspector_label][](plot, idx, pos)
    end
```

> At some point we changed this to be in the plots input space rather than world space (after model, transform_func) because that seems to be more useful with things like log plots.

Good point about log plots. For label data it's more useful to have access to the input space data.